sdk/core/src/main/java/org/apache/mesos/config/ConfigStore.java

```diff
@@ -26,14 +26,15 @@
     UUID store(T config) throws ConfigStoreException;
 
     /**
-     * Retrieves and deserializes the {@link Configuration} assigned to the provided UUID.
+     * Retrieves and deserializes the {@link Configuration} assigned to the provided UUID, or throws
+     * an exception if no config with the provided UUID was found.
      *
      * @param id The UUID of the configuration to be fetched
-     * @param factory The deserializer implementation for the configuration data
      * @return The deserialized configuration
-     * @throws ConfigStoreException if retrieval or deserialization fails
+     * @throws ConfigStoreException if retrieval or deserialization fails, or if the requested
+     *                              config is missing
      */
-    T fetch(UUID id, ConfigurationFactory<T> factory) throws ConfigStoreException;
+    T fetch(UUID id) throws ConfigStoreException;
```

Comment:
The factory param was originally because only this fetch() command actually used the factory. Moved it to the CuratorConfigStore constructor since:
- it's leaking a bit of implementation detail into the interface
- when are you going to want a write-only ConfigStore anyway? may as well just pass it to the constructor up-front.